packages/manager/modules/billing/src/autoRenew/actions/terminate-all-dom/controller.js

```diff
@@ -0,0 +1,129 @@
+  }
+
+  $onInit() {
+    this.isLoading = false;
+    this.cancelAllDom = true;
+  }
+
+  selectAllDomains() {
+    this.domains.forEach((domain) => set(domain, 'selected', true));
+  }
+
+  cancelServices() {
+    const domainsSelected = filter(this.domains, (domain) => domain.selected);
+    if (this.cancelAllDom || domainsSelected.length > 0) {
+      this.isLoading = true;
+      const servicesToTerminate = [];
+      if (this.cancelAllDom) {
+        servicesToTerminate.push(
+          this.$http
+            .get(`/allDom/${this.serviceId}/serviceInfos`)
+            .then(({ data }) => {
+              return new BillingService({
+                ...data,
+                serviceId: data.id,
+                serviceType: this.serviceType,
+              });
+            }),
+        );
+        this.trackCancel(this.serviceType);
+      }
+      domainsSelected.forEach((domain) => {
+        servicesToTerminate.push(
+          this.$http
+            .get(`/domain/${domain.name}/serviceInfos`)
+            .then(({ data }) => {
+              return new BillingService({
+                ...data,
+                serviceId: data.id,
+                serviceType: DOMAIN_SERVICE_TYPE,
+              });
+            }),
+        );
+        this.trackCancel(DOMAIN_SERVICE_TYPE);
+      });
+      this.$q
+        .all(servicesToTerminate)
+        .then((services) =>
+          map(services, (service) => {
+            service.setForResiliation();
+            return {
+              serviceId: service.domain,
+              serviceType: service.serviceType,
+              renew: service.renew,
+            };
+          }),
+        )
+        .then((services) => this.BillingAutoRenew.updateServices(services))
+        .then(() => this.onSuccess())
+        .catch((errorPromise) => {
+          this.$q.when(errorPromise).then((error) => {
+            if (error.messages) {
+              this.onPartialSuccess(error.messages);
+            } else {
+              this.onError(error);
+            }
+          });
+        });
+    } else {
+      this.goBack();
+    }
+  }
+
+  getService(serviceId, serviceType) {
+    return this.BillingAutoRenew.getService(serviceId, serviceType).then(
+      (service) =>
+        new BillingService({
+          ...service,
+          serviceId: service.id,
+        }),
+    );
+  }
+
+  trackCancel(serviceType) {
+    this.atInternet.trackClick({
+      name: `autorenew::${kebabCase(serviceType)}::delete::confirm`,
+      type: 'action',
+      chapter1: 'dedicated',
+      chapter2: 'account',
+      chapter3: 'billing',
+    });
+  }
+
+  onSuccess() {
+    this.goBack(this.$translate.instant('autorenew_all_dom_terminate_success'));
+  }
+
+  onPartialSuccess(errors) {
+    const messages = [];
+    messages.push(this.$translate.instant('autorenew_all_dom_terminate_error'));
+    errors.forEach((message) => {
+      messages.push(`<strong>${message.id}</strong>: ${message.message}`);
+    });
```

Comment:
```suggestion
    const messages = [
      this.$translate.instant('autorenew_all_dom_terminate_error'),
      errors.map((message) => `<strong>${message.id}</strong>: ${message.message}`),
    ];
```

Reply:
Map returns a new array. Having a map inside messages array adds a new array inside messages array. Which is not the intended behaviour.

Reply:
Yes sorry I forgot the spread operator, I meant:
```
const messages = [
      this.$translate.instant('autorenew_all_dom_terminate_error'),
      ...errors.map((message) => `<strong>${message.id}</strong>: ${message.message}`),
    ];
```